GoogleForCreators/web-stories-wp
# Completely offload large assets to CDN

## Summary

There's no strong reason to maintain large images (template assets, images used for plugin activation banner, FTUE story images (#4137)) within the plugin repository.

This change adapts the code base so that these are always loaded from `https://wp.stories.google/static/<version>`, where `version` is the current version number or `main` for nightly builds (any version number containing a commit hash)

**When working on new templates**

- Use your local dev environment's URL in the raw JSON files during development
- Once finished, update all URLs to use  `https://replaceme.com/images/templates/<templatename>/<filename>` 
  **Note:** there's a test ensuring this, so you'll notice any issues very quickly.
- Once finished, add all assets to the `static-site` branch, in the `public/static/main` folder

**Note:** eventually, the templates will be loaded from an API anyway, so this is just a small step in that direction.

## Relevant Technical Choices

- Don't host template assets on `gh-pages` anymore. Only using `static-site` from now on
- Always load template assets from `wp.stories.google`.
- New `WEBSTORIES_CDN_URL` constant
- For assets, use either CDN URL (always external) or assets URL (always internal)

## To-do

N/A

## User-facing changes

N/A

## Testing Instructions

- Check out branch or test this PR on the QA environment
- Explore templates
- Verify that all template images are loaded from `wp.stories.google`

---



Fixes #4015

See also #4137
